sotodlib/site_pipeline/make_ml_map.py

```diff
@@ -0,0 +1,322 @@
+            "nocal": True,
+            "nmat_dir": "/nmats",
+            "nmat_mode": "build",
+            "downsample": 1,
+            "maxiter": 500,
+            "tiled": 1,
+            "wafer": None,
+           }
+    
+def get_parser(parser=None):
+    if parser is None:
+        parser = ArgumentParser()
+    parser.add_argument("--config-file", type=str, default=None, 
+                     help="Path to mapmaker config.yaml file")
+
+    parser.add_argument("--query", type=str)
+    parser.add_argument("--freq", type=str)
+    parser.add_argument("--area", type=str)
+    parser.add_argument("--odir", type=str)
+    parser.add_argument("--prefix", type=str)
+    parser.add_argument('-C', "--comps",   type=str, help="T,Q, and/or U")
+    parser.add_argument("-c", "--context", type=str,  help="Context containing TODs")
+    parser.add_argument("-n", "--ntod",    type=int)
+    parser.add_argument(      "--tods",    type=str)
+    parser.add_argument(      "--nset",    type=int)
+    parser.add_argument("-N", "--nmat",    type=str)
+    parser.add_argument(      "--max-dets",type=int)
+    parser.add_argument("-S", "--site",    type=str)
+    parser.add_argument("-v", "--verbose", action="count")
+    parser.add_argument("-q", "--quiet",   action="count")
+    parser.add_argument("-@", "--center-at", type=str)
+    parser.add_argument("-w", "--window",  type=float)
+    parser.add_argument("-i", "--inject",  type=str)
+    parser.add_argument(      "--nocal",   action="store_true", default=True)
+    parser.add_argument(      "--nmat-dir", type=str)
+    parser.add_argument(      "--nmat-mode", type=str, help="How to build the noise matrix. 'build': Always build from tod. 'cache': Use if available in nmat-dir, otherwise build and save. 'load': Load from nmat-dir, error if missing. 'save': Build from tod and save.")
+    parser.add_argument("-d", "--downsample", type=int, help="Downsample TOD by this factor")
+    parser.add_argument(      "--maxiter",    type=int)
+    parser.add_argument("-T", "--tiled"  ,    type=int)
+    parser.add_argument("-W", "--wafer"  ,   type=str, nargs='+', help="Detector wafer subset to map with")
+    return parser
+
+
+def _get_config(config_file):
+    return yaml.safe_load(open(config_file,'r'))
+
+
+
+def main(config_file=None, defaults=defaults, **args):
+    
+    cfg = defaults
+    if config_file is not None:
+        # Load in config file
+        cfg_from_file = _get_config(config_file)
+        
+        # Check which fields have been supplied in config.yaml
+        for field, file_val in cfg_from_file.items():
+            # If an optional arg is supplied, in the config file
+            # add it here
+            if field not in cfg.keys():
+                cfg[field] = cfg_from_file[field]
+            # If the value in the config file is different from the 
+            # default, update it
+            elif cfg[field] != file_val:
+                cfg[field] = file_val
```

Comment:
I think lines 79-87 are equivalent to:
```
cfg.update(cfg_from_file)
```
Do you agree?

The only modification I might make, is to skip any settings from cfg_from_file, if they are None.  Because then a user setting "something: null" in their config file would be equivalent to saying "just use the default".

The one-liner for that is:
```
cfg.update({k: v for k, v in cfg_from_file.items() if v is not None})
```

Reply:
I'm guessing I can do something similar when merging values from the command line too (lines 91-94)

```   
# Merge flags from config file and defaults with any passed through CLI
    for key, item in args.items():
        if item is not None:
            cfg[key] = item
```
  Can then become:
  ```
  cfg.update({k: v for k, v in args.items() if v is not None})
  ```